Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SingleCrystalDiffuseReduction.py

```diff
@@ -197,134 +214,110 @@ def PyExec(self):
+                MDNorm(InputWorkspace='__bkg_md',
+                       FluxWorkspace='__flux',
+                       SolidAngleWorkspace='__sa',
+                       OutputDataWorkspace='__bkg_data',
+                       TemporaryDataWorkspace='__bkg_data' if mtd.doesExist('__bkg_data') else None,
+                       OutputNormalizationWorkspace='__bkg_norm',
+                       TemporaryNormalizationWorkspace='__bkg_norm' if mtd.doesExist('__bkg_norm') else None,
+                       OutputWorkspace='__normalizedBackground',
+                       QDimension0=self.getProperty('QDimension0').value,
+                       QDimension1=self.getProperty('QDimension1').value,
+                       QDimension2=self.getProperty('QDimension2').value,
+                       Dimension0Binning=self.getProperty('Dimension0Binning').value,
+                       Dimension1Binning=self.getProperty('Dimension1Binning').value,
+                       Dimension2Binning=self.getProperty('Dimension2Binning').value,
+                       SymmetryOperations=self.getProperty('SymmetryOperations').value)
+                DeleteWorkspace('__bkg_md')
+            progress.report()
             DeleteWorkspace('__run')
 
         if _background:
             # outWS = data / norm - bkg_data / bkg_norm * BackgroundScale
-            DivideMD(LHSWorkspace='__data',RHSWorkspace='__norm',OutputWorkspace=_outWS_name+'_normalizedData')
-            DivideMD(LHSWorkspace='__bkg_data',RHSWorkspace='__bkg_norm',OutputWorkspace=_outWS_name+'_normalizedBackground')
             CreateSingleValuedWorkspace(OutputWorkspace='__scale', DataValue=self.getProperty('BackgroundScale').value)
-            MultiplyMD(LHSWorkspace=_outWS_name+'_normalizedBackground',
+            MultiplyMD(LHSWorkspace='__normalizedBackground',
                        RHSWorkspace='__scale',
-                       OutputWorkspace='__scaled_background')
+                       OutputWorkspace='__normalizedBackground')
             DeleteWorkspace('__scale')
-            MinusMD(LHSWorkspace=_outWS_name+'_normalizedData',RHSWorkspace='__scaled_background',OutputWorkspace=_outWS_name)
+            MinusMD(LHSWorkspace='__normalizedData',RHSWorkspace='__normalizedBackground',OutputWorkspace=_outWS_name)
             if self.getProperty('KeepTemporaryWorkspaces').value:
                 RenameWorkspaces(InputWorkspaces=['__data','__norm','__bkg_data','__bkg_norm'],
                                  WorkspaceNames=[_outWS_name+'_data', _outWS_name+'_normalization',
                                                  _outWS_name+'_background_data',_outWS_name+'_background_normalization'])
         else:
             # outWS = data / norm
-            DivideMD(LHSWorkspace='__data',RHSWorkspace='__norm',OutputWorkspace=_outWS_name)
+            RenameWorkspace(InputWorkspace='__normalizedData',OutputWorkspace=_outWS_name)
```

Comment:
You could set directly the output workspace name in `MDNorm` for data, then optionally for `MinusMD` if you have background, so no need to rename